daemon/graphdriver/overlay/overlay.go

```diff
@@ -354,21 +351,34 @@ func (d *Driver) Get(id, mountLabel string) (_ containerfs.ContainerFS, err erro
 	if _, err := os.Stat(rootDir); err == nil {
 		return containerfs.NewLocalContainerFS(rootDir), nil
 	}
+
 	mergedDir := path.Join(dir, "merged")
 	if count := d.ctr.Increment(mergedDir); count > 1 {
 		return containerfs.NewLocalContainerFS(mergedDir), nil
 	}
 	defer func() {
 		if err != nil {
 			if c := d.ctr.Decrement(mergedDir); c <= 0 {
-				unix.Unmount(mergedDir, 0)
+				if mntErr := unix.Unmount(mergedDir, 0); mntErr != nil {
+					logrus.Debugf("Failed to unmount %s: %v: %v", id, mntErr, err)
+				}
+				if rmErr := os.RemoveAll(mergedDir); rmErr != nil {
+					logrus.Warnf("Failed to remove %s: %v: %v", id, rmErr, err)
+				}
```

Comment:
This should be a `Remove` rather than a `RemoveAll`, and I think there should be a comment akin to the one in `devicemapper` explaining why we explicitly do a `Remove` here. The same applies to all the other places you made this change.

Reply:
nit: `s/craeted/created/`